core/src/main/kotlin/com/exactpro/th2/read/db/impl/grpc/DataBaseReaderGrpcServer.kt

```diff
@@ -168,33 +170,59 @@ class DataBaseReaderGrpcServer(
     }
 
     private class GrpcExecuteListener(
-        private val observer: StreamObserver<QueryResponse>,
+        streamObserver: StreamObserver<QueryResponse>,
+        private val executionId: Long,
         private val event: Event,
         private val onEvent: (Event) -> Unit,
     ) : ResultListener {
+        private val skipped = AtomicLong()
+        private val observer = streamObserver as ServerCallStreamObserver<QueryResponse>
+
         override fun onRow(sourceId: DataSourceId, row: TableRow) {
-            requireNotNull(row.executionId) {
-                "'Execution id' is null for row: $row"
+            check(executionId == row.executionId) {
+                "'Execution id' isn't equal to '$executionId', row: $row"
+            }
+
+            while (!observer.isReady) {
+                if (observer.isCancelled) {
+                    skipped.incrementAndGet()
+                    return
+                }
```

Comment:
thought: I think this should be checked before everything

Reply:
I checked that onComplete doesn't throw an exception when request canceled